umap/static/umap/js/modules/data/layer.js

```diff
@@ -1040,14 +1073,17 @@ export class DataLayer {
     // Filename support is shaky, don't do it for now.
     const blob = new Blob([JSON.stringify(geojson)], { type: 'application/json' })
     formData.append('geojson', blob)
-    const saveUrl = this.map.urls.get('datalayer_save', {
+    console.log("map_id", this.map.options.umap_id, "pk", this.umap_id || this._future_uuid, "created", this.createdOnServer)
+    const saveURL = this.map.urls.get('datalayer_save', {
       map_id: this.map.options.umap_id,
-      pk: this.umap_id,
+      pk: this.umap_id || this._future_uuid,
+      created: this.createdOnServer,
```

Comment:
Instead of passing `created`, I'd better make here a condition and ask for `datalayer_update` or `datalayer_create`  depending on it